### PR TITLE
get liquidity distribution generic

### PR DESCRIPTION
**Pull request by silviutroscot**

### CODE REVIEW

File: packages/kamino-sdk/src/services/RaydiumService.ts [hunk 1]
```diff
@@ -24,8 +35,22 @@ export class RaydiumService {
...
+
+    let liqDistribution: LiquidityDistribution = {
+      data: [],
+    };
+    raydiumLiqDistribution.data.map((entry) => {
+      const liq: LiquidityForPrice = {
+        price: new Decimal(entry.price),
+        liquidity: new Decimal(entry.liquidity),
+      };
+      liqDistribution.data.push(liq);
```
Comment (peroxy) [inline packages/kamino-sdk/src/services/RaydiumService.ts:50]: This is a bit of a code smell - it's not ideal to push inside the map() function selector, hard to debug and find issues, it's not what you usually expect.

I would consider moving the push to liqDistribution to another line outside of the selector, or just replace map with forEach loop instead. Both of those are IMO better for this.

Comment (peroxy) [reply packages/kamino-sdk/src/services/RaydiumService.ts:50]: Same comment for the Orca method also

File: packages/kamino-sdk/src/utils/types.ts [hunk 2]
```diff
@@ -170,3 +170,12 @@ export interface VaultParameters {
   feeTier: Decimal;
   rebalancingParameters: RebalanceFieldInfo[];
 }
+
+export interface LiquidityDistribution {
+  data: LiquidityForPrice[];
```
Comment (peroxy) [inline packages/kamino-sdk/src/utils/types.ts:175]: I think `data` naming is a bit too generic, I would consider renaming this to liquidities, distributions, liquidityDistributions or something similar. Easier to understand when reading the code that uses this.

**Review by peroxy — CHANGES_REQUESTED** [review]
Looks good, have a few minor comments.

Will this only be used when creating a strategy, so you will only need to request one pool's data? Would you ever use this to batch request all pool data? In that case it might be useful to batch fetch and make it a bit more optimized.

**Comment by silviutroscot** [comment]
> Looks good, have a few minor comments.
> 
> Will this only be used when creating a strategy, so you will only need to request one pool's data? Would you ever use this to batch request all pool data? In that case it might be useful to batch fetch and make it a bit more optimized.

yes, afaik it will be used only on strat creation, @egor-humandone can you confirm?

**Review by peroxy — APPROVED** [review]

**Comment by y2kappa** [comment]
> > Looks good, have a few minor comments.
> > Will this only be used when creating a strategy, so you will only need to request one pool's data? Would you ever use this to batch request all pool data? In that case it might be useful to batch fetch and make it a bit more optimized.
> 
> yes, afaik it will be used only on strat creation, @egor-humandone can you confirm?

@egor-humandone doesn't reply here, the conversation rarely escapes this thread to people not involved in the PR, fyi.

If you want more info/help ask in discord.

I happened to see this comment by chance.

**Pull request merged**